terratorch/datasets/open_sentinel_map.py

```diff
@@ -0,0 +1,222 @@
+        ax[-1].axis("off")
+
+        if suptitle:
+            plt.suptitle(suptitle)
+
+        return fig
+
+    def __getitem__(self, index: int) -> dict[str, torch.Tensor]:
+        mgrs_tile, spatial_cell = self.image_files[index]
+        spatial_cell_path = self.imagery_root / mgrs_tile / spatial_cell
+
+        npz_files = list(spatial_cell_path.glob("*.npz"))
+        npz_files.sort(key=lambda x: self._extract_date_from_filename(x.stem))
+
+        if self.pick_random_pair:
+            npz_files = random.sample(npz_files, 2)
+            npz_files.sort(key=lambda x: self._extract_date_from_filename(x.stem))
+
+        output = {}
+
+        if self.interpolate_and_concat_bands:
+            images_over_time = []
+            for _, npz_file in enumerate(npz_files):
+                data = np.load(npz_file)
+                interpolated_bands = []
+                for band in self.bands:
+                    band_frame = data[band]
+                    band_frame = torch.from_numpy(band_frame).float()
+                    band_frame = band_frame.permute(2, 0, 1)
+                    interpolated = F.interpolate(
+                        band_frame.unsqueeze(0), size=(192, 192), mode="bilinear", align_corners=False
+                    ).squeeze(0)
+                    interpolated_bands.append(interpolated)
+                concatenated_bands = torch.cat(interpolated_bands, dim=0)
+                images_over_time.append(concatenated_bands)
+
+            images = torch.stack(images_over_time, dim=0).numpy()
+            if self.truncate_image:
+                images = images[-self.truncate_image:]
+            if self.pad_image:
+                images = pad_numpy(images, self.pad_image)
+
+            output["image"] = images.transpose(0, 2, 3, 1)
+        else:
+            image_dict = {band: [] for band in self.bands}
+            for _, npz_file in enumerate(npz_files):
+                data = np.load(npz_file)
+                for band in self.bands:
+                    band_frames = data[band]
+                    band_frames = band_frames.astype(np.float32)
+                    band_frames = np.transpose(band_frames, (2, 0, 1))
+                    image_dict[band].append(band_frames)
+
+            final_image_dict = {}
+            for band in self.bands:
+                band_images = image_dict[band]
+                if self.truncate_image:
+                    band_images = band_images[-self.truncate_image:]
+                if self.pad_image:
+                    band_images = [pad_numpy(img, self.pad_image) for img in band_images]
+                band_images = np.stack(band_images, axis=0)
+                final_image_dict[band] = band_images
+
+            output["image"] = final_image_dict
+
+        label_file = self.label_files[index]
+        mask = np.array(Image.open(label_file)).astype(int)
+        mask[mask == 254] = 15  # noqa: PLR2004
+        mask[mask == 255] = 16  # noqa: PLR2004
```

Comment:
This was done because the _unlabel_ and _none_ categories are represented by 254 and 255 in the mask, while the other categories range from 0 to 14. These categories were likely set as 254 and 255 to be used as ignore_index, but in other cases, it might be useful to include them. That's why I mapped them to unused category indices. Will clarify this further in the comments.